components/Pagination/Pagination.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { number, string } from 'prop-types';
+import classNames from 'classnames';
+import LeftAngleIcon from 'static/images/icons/FontAwesome/angle-left-solid.svg';
+import RightAngleIcon from 'static/images/icons/FontAwesome/angle-right-solid.svg';
+import PaginationItem from './PaginationItem/PaginationItem';
+import styles from './Pagination.css';
+
+Pagination.propTypes = {
+  className: string,
+  currentPage: number.isRequired,
+  totalPages: number.isRequired,
+};
+
+Pagination.defaultProps = {
+  className: undefined,
+};
+
+const getPagination = (currentPage, totalPages) => {
+  // maximum length of the Pagination Bar, should be an odd integer, delault is 11
+  const MAX_VISIBLE_ELEMENTS = 11;
+  const ELEMENTS_ON_ONE_SIDE = (MAX_VISIBLE_ELEMENTS - 1) / 2; // 5
+
+  const isTruncatingRequired = totalPages >= MAX_VISIBLE_ELEMENTS;
+
+  const isLeftSideLengthy = currentPage - 1 > ELEMENTS_ON_ONE_SIDE;
+  const isRightSideLengthy = totalPages - currentPage > ELEMENTS_ON_ONE_SIDE;
+
+  const shouldTruncateStart = isTruncatingRequired && isLeftSideLengthy;
+  const shouldTruncateEnd = isTruncatingRequired && isRightSideLengthy;
+
+  const truncateStartOnly = shouldTruncateStart && !shouldTruncateEnd;
+  const truncateEndOnly = !shouldTruncateStart && shouldTruncateEnd;
+
+  let paginationStart;
+  let paginationEnd;
+
+  if (truncateStartOnly) {
+    paginationStart = totalPages - MAX_VISIBLE_ELEMENTS + 3;
+    paginationEnd = totalPages;
+  } else if (truncateEndOnly) {
+    paginationStart = 1;
+    paginationEnd = MAX_VISIBLE_ELEMENTS - 2;
+  } else {
+    paginationStart = shouldTruncateStart ? currentPage - (ELEMENTS_ON_ONE_SIDE - 2) : 1;
+    paginationEnd = shouldTruncateEnd ? currentPage + (ELEMENTS_ON_ONE_SIDE - 2) : totalPages;
+  }
+
+  const paginationLength = paginationEnd - paginationStart + 1;
+
+  return {
+    paginationStart,
+    paginationLength,
+    shouldTruncateStart,
+    shouldTruncateEnd,
+  };
+};
+
+// eslint-disable-next-line react/prop-types
+const PaginationItems = ({ currentPage, totalPages }) => {
```

Comment:
You can drop the lint disable and delete `className` as a possible prop!

Reply:
I guess if we drop the lint disable it would require us to define prop types for currentPage and totalPages, but they are already verified when we require them in `Pagination` component.

Reply:
Ah it's cuz two JSX methods in one file. Okay, no worries!